metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -306,4 +308,24 @@ final class Diagnostics(
     toPublish
   }
 
+  private def adjustDiagnosticData(
+      diagnostic: l.Diagnostic,
+      edit: TokenEditDistance,
+  ): Object =
+    diagnostic match {
+      case ScalacDiagnostic.ScalaDiagnostic(Left(textEdit)) =>
+        val range = textEdit.getRange
+        val revisedRange = edit
+          .toRevised(
+            range = range,
+            adjustWithinToken = shouldAdjustWithinToken(diagnostic),
+          )
+        textEdit.setRange(revisedRange.getOrElse(range))
+        textEdit.toJsonObject
+      // TODO also update bsp scala diagnostics, e.g. ScalaDiagnostic(Right(textEdits))
+      case diagnostic => diagnostic.getData
```

Comment:
Currently, no token match means we set the diagnostic data of the fresh diagnostic to the old, possibly out of date data.  This maintains the current behaviour, though I was thinking it might make more sense just to set the data as empty so the out of date code action doesn't appear?

Reply:
> I was thinking it might make more sense just to set the data as empty so the out of date code action doesn't appear

Hmm... that might be a good idea. 

Btw. Did you find any issues with updating also the `Right` branch of ScalaDiagnostic? We will be deprecating just sending TextEdit with the next ScalaCLI version, so would probably be good to do both. I can work on that if you don't have time.

Reply:
Yeah it looks something similar can be done, happy to include it in this PR if you're ok with that :+1: .  Is there anything using the new `ScalaDiagnostic` I can use to test it out?

Reply:
That would be cool! I think you can test it out with the latest nightly of ScalaCLI. `scala-cli --cli-version nightly setup-ide .  ` should work I think on a sample project.

Reply:
Thanks! I had to change `"scala"` to `"scala-cli"`, but it looks to have done the trick.